lib/howl/commands/app_commands.moon

```diff
@@ -302,36 +313,54 @@ command.register
 -----------------------------------------------------------------------
 
 launch_cmd = (working_directory, cmd) ->
-  shell = howl.sys.env.SHELL or '/bin/sh'
-  p = Process {
-    :cmd,
-    :shell,
-    read_stdout: true,
-    read_stderr: true,
-    working_directory: working_directory,
-  }
-
-  buffer = ProcessBuffer p
-  editor = app\add_buffer buffer
-  editor.cursor\eof!
-  buffer\pump!
+  dispatch.launch ->
```

Comment:
I believe I see the reason for doing it like this, with the stack and all and the need for a return to properly deactivate things. I'm not sure it sounds it's a good idea though to require all command handlers to be aware of this. For one, I imagine this is really the most common mode for most commands; they might perhaps read some input but after that they're going to do something else, which might or might not end up yielding in some operation. Whether a certain operation will yield or not is not something that will always be readily apparent, and might also change after the command has been written if the implementation of something else changes. The nice thing about having a coroutine-based API, such as for commands now, and for other things in the future, is that you can just write your code and not worry to much about whether this will block or not, etc. This seems a step backwards in that regard. To be on the safe side commands would then always have to explicitly launch stuff in an co-routine. That stuff is not trivial to get right though. Consider for instance the example above: `dispatch.launch` is invoked, but nothing checks the return value, so any errors before the actual yield later is silently ignored.

It would be better if this could be supported automatically. What would the issues be if all handlers were dispatched in their own coroutines? It would be possible to know after a dispatch whether the handler is done or has yielded, so that information could then be used for determining how to handle the command line logic.